pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -729,24 +730,28 @@ func updateStatefulSetAfterInvariantEstablished(
 		if err != nil {
 			return &status, err
 		}
+		metrics.MaxUnavailable.WithLabelValues(set.Namespace, set.Name, podManagementPolicy).Set(float64(maxUnavailable))
 	}
 
 	// Collect all targets in the range between getStartOrdinal(set) and getEndOrdinal(set). Count any targets in that range
 	// that are unavailable. Select the
 	// (MaxUnavailable - Unavailable) Pods, in order with respect to their ordinal for termination. Delete
 	// those pods and count the successful deletions. Update the status with the correct number of deletions.
 	unavailablePods := 0
+
 	for target := len(replicas) - 1; target >= 0; target-- {
 		if isUnavailable(replicas[target], set.Spec.MinReadySeconds) {
 			unavailablePods++
 		}
 	}
+	metrics.UnavailableReplicas.WithLabelValues(set.Namespace, set.Name, podManagementPolicy).Set(float64(unavailablePods))
 
 	if unavailablePods >= maxUnavailable {
-		logger.V(2).Info("StatefulSet found unavailablePods, more than or equal to allowed maxUnavailable",
+		logger.V(4).Info("StatefulSet found unavailablePods, more than the allowed maxUnavailable",
```

Comment:
The current log message is triggered even when `unavailablePods == maxUnavailable`, which is not a violation but an expected state during a rolling update.

To avoid logging during normal operations, what do you think about separating the logic? We could log only when there's a true violation (`unavailablePods > maxUnavailable`) while still returning when `unavailablePods >= maxUnavailable`. This would make the logs more accurate.